src/fmeval/eval_algorithms/answer_relevance.py

```diff
@@ -0,0 +1,289 @@
+
+    :param strictness: The number questions generated per answer.
+    """
+
+    strictness: Optional[int] = 1
+
+
+class AnswerRelevanceScore(Transform):
+    """This transform augments its input record with the computed answer relevance score.
+
+    See the docstring for `Answer Relevance` for more details regarding the score itself.
+    """
+
+    def __init__(
+        self,
+        embeddings_model: ModelRunner,
+        model_output_key: str = DatasetColumns.MODEL_OUTPUT.value.name,
+        output_key: str = ANSWER_RELEVANCE,
+    ):
+        """AnswerRelevanceScore initializer.
+
+        :param embeddings_model: An instance of ModelRunner representing the embedding model to be used.
+        :param model_output_key: The key corresponding to the model output.
+        :param output_key: The key corresponding to the answer relevance score that will be added to the input record.
+        """
+        super().__init__(embeddings_model, model_output_key, output_key)
+        self.register_input_output_keys(
+            input_keys=["gen_questions", model_output_key],
+            output_keys=[output_key],
+        )
+        self.output_key = output_key
+        self.model_output_key = model_output_key
+        self.embeddings_model = embeddings_model
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Augment the input record with the computed answer relevance score.
+
+        :param record: The input record.
+        :returns: The input record, with the answer relevance score added in.
+        """
+        gen_questions = record["gen_questions"].split("\n")
+        question = record[self.model_output_key]
+        record[self.output_key] = self._get_score(self.embeddings_model, question, gen_questions)
+        return record
+
+    @staticmethod
+    def _get_score(embeddings_model: ModelRunner, question: str, generated_questions: List[str]) -> float:
+        """Given generated questions and the original question, embedding model, computer Answer Relevance score.
+
+        :param embeddings_model: An instance of ModelRunner representing the embedding model to be used.
+        :param question: The given prompt (model input).
+        :param generated_questions: An artificial question generated by judge model based on the answer.
+        :returns: 0 to 1. See the docstring for `Answer Relevance` for more details
+            on what these numerical values represent.
+        """
+        question_vector = np.asarray(embeddings_model.predict(question))
+        gen_question_vectors = []
+        for generated_question in generated_questions:
+            gen_question_vector = np.asarray(embeddings_model.predict(generated_question))
+            gen_question_vectors.append(gen_question_vector)
+        norm = np.linalg.norm(gen_question_vectors) * np.linalg.norm(question_vector)
+        return np.dot(gen_question_vector, question_vector.T) / norm
```

Comment:
This doesn't seem like it computes pairwise dot product between each generated question embedding and original question embedding. Also missing the mean computation.

Reply:
@awsvmaringa mean computation will be applied when aggregate the dataset score. This `_get_score` method is per record.

Reply:
I believe he's referring to the fact that the formula here is using `gen_question_vector`, which is just one of many generated question vectors. Ideally, we would be referencing a formula from a paper or something; what have you been using @xiaoyi-cheng?

If we're not using a reference, I propose these two options.

1) Compute the mean of the generated question vectors and then return the normalized dot product of the mean vector with the question vector.
In code: `return np.dot(mean_gen_question_vector, question_vector) / norm`

2) Compute the normalized dot product of each `(gen_question_vector, question_vector)` pairing (where the only quantity that's changing is `gen_question_vector`) and then average these scalar values for your final "score". Since you'd need to normalize each of the vectors in your matrix independently, you'd have to tweak the current code, as `norm` wouldn't be useful anymore (as mentioned in my comment above, it's the Frobenius norm of the whole matrix).

Reply:
We're implementing this [metric](https://docs.ragas.io/en/stable/concepts/metrics/answer_relevance.html) for reference.

Reply:
Thanks! Looks like it's just option 2 that I mentioned above.